scripts/irods/test/test_ichmod.py

```diff
@@ -28,6 +29,180 @@ def test_ichmod_r_with_garbage_path(self):
         self.user.assert_icommand('iget ' + self.admin.session_collection + '/sub_dir1\\\\%/subdir2/file ' + os.path.join(self.user.local_session_dir, ''))
 
+    def test_ichmod_access_levels(self):
+        filename1 = 'test_ichmod_access_levels'
+        filename2 = 'test_access_levels_put_file'
+        local_file_path = os.path.join(self.admin.local_session_dir, filename1)
+        public_collection = os.path.join('/' + self.admin.zone_name, 'home', 'public')
+
+        test_object = os.path.join(public_collection, filename1)
+        test_collection_for_put = os.path.join(public_collection, 'test_ichmod_collection_put')
+        test_collection_for_access = os.path.join(public_collection, 'test_ichmod_collection_access')
+        test_collection_for_create = os.path.join(test_collection_for_put, 'new_collection')
+
+        user_file_path = os.path.join(self.user.local_session_dir, filename2)
+        user_iget_file = os.path.join(self.user.local_session_dir, filename1)
+
+        coll_access_path = os.path.join(test_collection_for_put, filename1)
+
+        class Access(Enum):
+            NULL            = 1
+            READ_METADATA   = 2
+            READ_OBJECT     = 3
+            CREATE_METADATA = 4
+            MODIFY_METADATA = 5
+            DELETE_METADATA = 6
+            CREATE_OBJECT   = 7
+            MODIFY_OBJECT   = 8
+            DELETE_OBJECT   = 9
+            OWN             = 10
+
+        access_specifiers = {
+            Access.NULL            : "null",        
+            Access.READ_METADATA   : "read metadata",    
+            Access.READ_OBJECT     : "read object",
+            Access.CREATE_METADATA : "create metadata",
+            Access.MODIFY_METADATA : "modify metadata",
+            Access.DELETE_METADATA : "delete metadata", 
+            Access.CREATE_OBJECT   : "create object",
+            Access.MODIFY_OBJECT   : "modify object",
+            Access.DELETE_OBJECT   : "delete object",
+            Access.OWN             : "own"
+        }
+
+        # Here the test commands we test with every acces level are defined:
+        # 'cmd' is the command to run with assert_icommand
+        # the 'checks' list contains an access level range and the output of the command to match
+        # in the format (<min_level>, <max_level>, <arguments>)
+        # *arguments will be passed to the assert_icommands function
+        #
+        # 
+        test_commands = {
+            # Test that iget needs READ OBJECT
+            'iget': {
+                'cmd': ['iget' , '-f', test_object, user_iget_file],
+                'checks': [
+                    (Access.NULL, Access.READ_METADATA,  ['STDERR', 'USER_INPUT_PATH_ERR']),
+                    (Access.READ_METADATA, Access.OWN,   ['STDERR', ''])
+                ]
+            },
+            # Test that iput needs MODIFY_OBJECT on the collection
+            # NB why is CREATE OBJECT not enough?
```

Comment:
Agreed that this seems wrong. This is likely an artifact of the fact that the interface for changing permissions has been the way that it is, and the reason this change exists.